salalib/mgraph.cpp

```diff
@@ -1299,6 +1299,30 @@ bool MetaGraph::analyseSegmentsAngular( Communicator *communicator, Options opti
    return retvar;
 }
 
+bool MetaGraph::analyseTopoMetMultipleRadii( Communicator *communicator, Options options ) // <- options copied to keep thread safe
+{
+   m_state &= ~SHAPEGRAPHS;      // Clear axial map data flag (stops accidental redraw during reload)
+
+   bool retvar = false;
+
+   try {
+      // note: "output_type" reused for analysis type (either 0 = topological or 1 = metric)
+      retvar = true;
+      for(size_t i = 0; i < options.radius_list.size(); i++) {
+          if(!getDisplayedShapeGraph().analyseTopoMet(communicator, options.output_type, options.radius_list[i], options.sel_only)) {
+              retvar = false;
+          }
+      }
+   }
+   catch (Communicator::CancelledException) {
+      retvar = false;
+   }
+
+   m_state |= SHAPEGRAPHS;

```

Comment:
Is this even correct? If the analysis has been cancelled, should we update the state?